R-package/tests/testthat/test_utils.R

```diff
@@ -73,9 +73,9 @@ test_that("lgb.last_error() correctly returns errors from the C++ side", {
         data = train$data
         , label = as.matrix(rnorm(5L))
     )
-    # expect_error({
-    #     dvalid1$construct()
-    # }, regexp = "[LightGBM] [Fatal] Length of label is not same with #data", fixed = TRUE)
+    expect_error({
+        dvalid1$construct()
+    }, regexp = "[LightGBM] [Fatal] Length of label is not same with #data", fixed = TRUE)
 })
```

Comment:
ok I can tell from testing that this specific test causes this error:

```text
==1313== 
==1313== HEAP SUMMARY:
==1313==     in use at exit: 264,131,790 bytes in 46,136 blocks
==1313==   total heap usage: 2,436,607 allocs, 2,390,471 frees, 5,670,514,616 bytes allocated
==1313== 
==1313== 20 bytes in 1 blocks are definitely lost in loss record 29 of 2,467
==1313==    at 0x483BE63: operator new(unsigned long) (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==1313==    by 0x151880A1: __gnu_cxx::new_allocator<float>::allocate(unsigned long, void const*) (new_allocator.h:115)
==1313==    by 0x15186DE9: std::allocator_traits<std::allocator<float> >::allocate(std::allocator<float>&, unsigned long) (alloc_traits.h:460)
==1313==    by 0x151855C3: std::_Vector_base<float, std::allocator<float> >::_M_allocate(unsigned long) (stl_vector.h:346)
==1313==    by 0x15183CE2: std::_Vector_base<float, std::allocator<float> >::_M_create_storage(unsigned long) (stl_vector.h:361)
==1313==    by 0x15180976: std::_Vector_base<float, std::allocator<float> >::_Vector_base(unsigned long, std::allocator<float> const&) (stl_vector.h:305)
==1313==    by 0x152959AC: std::vector<float, std::allocator<float> >::vector(unsigned long, std::allocator<float> const&) (stl_vector.h:511)
==1313==    by 0x15444AB8: LGBM_DatasetSetField_R (lightgbm_R.cpp:220)
==1313==    by 0x4941BDA: R_doDotCall (dotcode.c:610)
==1313==    by 0x494CCC6: do_dotcall (dotcode.c:1281)
==1313==    by 0x499FB01: bcEval (eval.c:7078)
==1313==    by 0x498B67F: Rf_eval (eval.c:727)
==1313== 
==1313== LEAK SUMMARY:
==1313==    definitely lost: 20 bytes in 1 blocks
==1313==    indirectly lost: 0 bytes in 0 blocks
==1313==      possibly lost: 0 bytes in 0 blocks
```

I'm going to try to push a fix